crates/host/src/wasmbus/mod.rs

```diff
@@ -1263,15 +1263,14 @@ impl Host {
                                                 tasks.spawn(async move {
                                                     let _permit = permit;
                                                     debug!("handling invocation");
-                                                    match fut.await {
-                                                        Ok(()) => {
-                                                            debug!("successfully handled invocation");
-                                                            Ok(())
+                                                    match timeout(max_execution_time, fut).await {
+                                                        Ok(Err(err)) => {
+                                                            error!(?err, "component handler error");
                                                         },
                                                         Err(err) => {
-                                                            warn!(?err, "failed to handle invocation");
-                                                            Err(err)
+                                                            error!(?err, "component handler timed out");
                                                         },
+                                                        Ok(_) => {},
```

Comment:
small nit, I think it would be good to keep the trace event here for a successful invocation just to confirm 😄 
```suggestion
                                                        Ok(_) => {
                                                          debug!("successfully handled invocation")
                                                        },
```